solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java

```diff
@@ -141,6 +134,16 @@ public int getId() {
     return id;
   }
 
+  /**
+   * @param id
+   *          Unique ID for this feature. Similar to feature name, except it can
+   *          be used to directly access the feature in the global list of
+   *          features.
+   */
+  public void setId(int id) {
```

Comment:
What's the reason for ripping the id and not keeping it together with the name in the constructor?